tf_agents/replay_buffers/episodic_replay_buffer.py

```diff
@@ -896,8 +896,9 @@ def _maybe_end_episode_id():
       """Maybe end episode ID."""
       def _end_episode_id():
         return tf.group(
-            tf.compat.v1.scatter_update(self._episode_completed,
-                                        [episode_location], 1))
+          self._episode_completed.scatter_update(tf.IndexedSlices(tf.ones_like(
+            episode_location, dtype=self._episode_completed.dtype),
```

Comment:
I think I prefer the old API.  Do we really need IndexedSlices?

Reply:
As far as I understand it, all of the `scatter_*` methods on the TensorFlow `Variable` use `IndexedSlices` now. I propose changing to this API because it uses the public interface of a `Variable` for the update, as opposed to the implementation of the v1 `scatter_update` function that relies on protected methods. 

I assume that protected methods are not guaranteed to be implemented on the `MirroredVariable`, however I am happy to be corrected if I have misunderstood anything!

Reply:
@ebrevdo back to you.

Reply:
Ok but why ones_like?  Shouldn't it just be episode_location?

Reply:
That appeared to maintain the functionality. 

I am not sure that I understand how using the values in `episode_location` is supposed to work. It appears that the entries in `_episode_complete` should be greater than zero at the indices of steps at the ends of episodes. As such, using the values of `episode_location` would have no effect where the index was equal to zero. In particular, using `episode_location` appears to fail some unit tests.

Reply:
Bummer.  Sorry for the delay.  I'll test and see why unit tests are failing in that case.

Reply:
I tried some tweaks.  change all tf.zeros_like and tf.ones_like in the IndexedSlices value args to just 0 and 1.  This worked for me and all tests passed.

I think once that's done, we can approve this.

Reply:
@leroidauphin lmk if that change causes errors for you.  it worked fine for me.

Reply:
@ebrevdo I have made the change. I apologise for my mistake.